app/src/main/java/com/example/booklisting/MainActivity.java

```diff
@@ -130,38 +125,42 @@ public void onLoadMore() {
     }
 
     @Override
-    public void onLoadFinished(@NonNull Loader<List<Book>> loader, List<Book> list) {
+    public void onLoadFinished(@NonNull Loader<BookLoaderResult> loader, BookLoaderResult loaderResult) {
         if ((adapter == null) || (adapter.getItemCount() == 0)) {
-            if (exceptionThrown) {
+            if (loaderResult.getException() != null) {
+                Log.i("Exception is", String.valueOf(loaderResult.getException()));
```

Comment:
Log code should be used only for local test, it should not be committed or merged to master

Reply:
Corrected.